test/boost/cql_query_test.cc

```diff
@@ -150,6 +150,98 @@ SEASTAR_TEST_CASE(test_twcs_max_window) {
+        e.execute_cql("CREATE TABLE tbl7 (a int PRIMARY KEY, b int) WITH compaction = "
+                      "{'class': 'TimeWindowCompactionStrategy'} AND default_time_to_live = 86400000;").get();
+        e.require_table_exists("ks", "tbl7").get();
+        e.execute_cql("UPDATE system.config SET value='50' WHERE name='twcs_max_window_count';").get();
+        e.execute_cql("ALTER TABLE tbl7 WITH compaction = {'class': 'SizeTieredCompactionStrategy'}").get();
+
+        // Cenario 8: No TTL TWCS table to STCS
+        e.execute_cql("CREATE TABLE tbl8 (a int PRIMARY KEY, b int) WITH compaction = "
+                      "{'class': 'TimeWindowCompactionStrategy'};").get();
+        e.require_table_exists("ks", "tbl8").get();
+        e.execute_cql("ALTER TABLE tbl8 WITH compaction = {'class': 'SizeTieredCompactionStrategy'};").get();
+
+        // Cenario 9: Large TTL TWCS table, modify attribute other than compaction and default_time_to_live
+        e.execute_cql("UPDATE system.config SET value='0' WHERE name='twcs_max_window_count';").get();
+        e.execute_cql("CREATE TABLE tbl9 (a int PRIMARY KEY, b int) WITH compaction = "
+                      "{'class': 'TimeWindowCompactionStrategy'} AND default_time_to_live = 86400000;").get();
+        e.require_table_exists("ks", "tbl9").get();
+        e.execute_cql("UPDATE system.config SET value='50' WHERE name='twcs_max_window_count';").get();
+        e.execute_cql("ALTER TABLE tbl9 WITH gc_grace_seconds = 0;").get();
+
+        // Cenario 10: Large TTL STCS table, fail to switch to TWCS with no TTL
+        e.execute_cql("CREATE TABLE tbl10 (a int PRIMARY KEY, b int) WITH default_time_to_live = 8640000;").get();
+        e.require_table_exists("ks", "tbl10").get();
+        assert_that_failed(e.execute_cql("ALTER TABLE tbl10 WITH compaction = {'class': 'TimeWindowCompactionStrategy'};"));
+        e.execute_cql("ALTER TABLE tbl10 WITH compaction = {'class': 'TimeWindowCompactionStrategy'} AND default_time_to_live = 0;").get();
+
+        // Cenario 11: Ensure default_time_to_live updates reference existing table properties
+        e.execute_cql("CREATE TABLE tbl11 (a int PRIMARY KEY, b int) WITH compaction = "
+                      "{'class': 'TimeWindowCompactionStrategy', 'compaction_window_size': '1', "
+                      "'compaction_window_unit': 'MINUTES'} AND default_time_to_live=3000;").get();
+       e.require_table_exists("ks", "tbl11").get();
+       assert_that_failed(e.execute_cql("ALTER TABLE tbl11 WITH default_time_to_live=3600;"));
+       e.execute_cql("ALTER TABLE tbl11 WITH compaction = {'class': 'TimeWindowCompactionStrategy', "
+                     "'compaction_window_size': '2', 'compaction_window_unit': 'MINUTES'};").get();
+       e.execute_cql("ALTER TABLE tbl11 WITH default_time_to_live=3600;").get();
+
+       // Cenario 12: Ensure that window sizes <= 0 are forbidden
```

Comment:
Nitpick: you added 12 scenarios not 10 as the commit message says, and mis-spelled "scenario" in all of them ;-)